invenio_app_rdm/records_ui/views/deposits.py

```diff
@@ -78,6 +78,89 @@ def get_form_pids_config():
     return pids_providers
 
 
+class ExternalResourcesParser:
+    """Entry class for parsing records into their appropriate UI components."""
+
+    def __init__(self, resources):
+        """Constructor."""
+        self.resources = resources
+
+    def _parse_external_resources(self):
+        """Entry parser function for serializing resources based on their scheme."""
+        resources = []
+
+        for resource in self.resources:
+            scheme = resource["scheme"]
+
+            # Only append elenents if we have a 'parser' for it
+            if hasattr(ResourceParser, scheme):
+                parser = ResourceParser(resource)
+                item = getattr(parser, scheme)()
+
+                resources.append(item)
+        return resources
+
+    def dump(self):
+        """Dumps the resource into a dictionary to later be treated by the UI."""
+        return self._parse_external_resources()
+
+
+class ResourceParser:
+    """Parser for different scheme types."""
+
+    def __init__(self, resource):
+        """Constructor."""
+        self.resource = resource
+
+    def url(self):
```

Comment:
It gets called here:

https://github.com/TLGINO/invenio-app-rdm/blob/1aae1a016e2e79e3c6681a0379df28aaf1dfd057/invenio_app_rdm/records_ui/views/deposits.py#L96

```
if hasattr(ResourceParser, scheme):
    parser = ResourceParser(resource)
    item = getattr(parser, scheme)()
```

This is done with the idea that we may want to parse different types of `schema` and not just `url`
In which caase it would suffice to add another method to `ResourceParser` with a name corresponding to it's `schema`